test/util/http.js

```diff
@@ -253,9 +270,48 @@ is equivalent to:
     .route('/')
     .mount(App, { router })
 
-A component may send a request but not be associated with a route, for example,
-many modals. To test a component like that, use mockHttp(), or use load() to
-mount the component's parent component.
+Which of these approaches should I take?
+----------------------------------------
+
+Every test of a component will need to mount the component. There are a few ways
+to do so:
+
+  1. mount()
+  2. mockHttp()
+  3. load()
+  4. load(), specifying `root` as `false`
```

Comment:
I suggest reordering this list and having the most common/straightforward ones at the top. It also matches the explanations below. 

```
  1. load() - used when loading the entire App is necessary
  2. mount() - used for testing a component in isolation (fastest)
  3. load(), specifying `root` as `false`
  4. mockHttp()
```

Reply:
I actually think that in some ways, the current order already goes from most straightforward to least straightforward. It starts with `mount()`, which comes from Vue Test Utils and only mounts a component. It then moves on to `mockHttp()`, which mocks `$http` and can also mount a component by calling `mount()`. After that is `load()`, which calls `mockHttp()`, then specifies responses.

I'm not sure that this section is successful at doing so, but what I want is to set up `mount()` and `mockHttp()` as the defaults while also presenting `load()` as the go-to function in certain circumstances. `mount()` and `mockHttp()` are more explicit and do less, and now that we can mock `$route`, we should be able to use `mount()` and `mockHttp()` in more tests. Yet if you need to mount `App`, or if you need to mount a route component that sends a request, then `load()` is definitely the most convenient option.

Let me know if you can think of changes along those lines that we could make to this section! Like maybe we should say here that `mockHttp()` calls `mount()`, and `load()` calls `mockHttp()` — maybe that would give a sense of increasing complexity? I also think your comment that `mount()` is fastest is helpful — I'll add that to this section now.